crates/circuit/src/circuit_data.rs

```diff
@@ -159,7 +158,7 @@ impl CircuitData {
             qubit_indices: BitLocator::new(),
```

Comment:
The `BitLocator` has a `cached: OnceLock<Py<PyDict>>`, can this become an issue at runtime or is this fine since there's no way we can populate the cache with Python objects?

Reply:
It shouldn't be a problem we should only be initializing or working with the cached attribute when returning to python. There shouldn't be any code paths we access from cext where we could use this (or `BitLocator` more broadly tbh). We can refactor more if we do encounter issues though.